Genius/Src/main.c

```diff
@@ -179,8 +180,8 @@ int main(void)
   MX_ADC1_Init();
   MX_DAC1_Init();
   MX_UART5_Init();
-  MX_USB_OTG_FS_PCD_Init();
-  MX_USB_OTG_HS_HCD_Init();
+  //MX_USB_OTG_FS_PCD_Init();
+  //MX_USB_OTG_HS_HCD_Init();

```

Comment:
how come they are commented out?

Reply:
Commented out because firmware hangs if it's enabled. Was going to ask for any suggestions why this could happen, I might be missing something in bootloader.

Also (not sure if related), Cube was generating HAL_PCD_MspInit function that is also defined in MidiBoot/usbd-*s/ - I assumed that cube's function is supposed to be replaced by our version in this case?

Reply:
Should probably use the Cube version, it should be the same but more up to date.
They keep moving and renaming things don't they!
HAL_PCD_Init used to be called in usbd_conf.c from USBD_LL_Init. Hold on, it's still there...
Are those two functions not meant to be generated if there's no device class specified?
Otherwise I think the init will happen when `MX_USB_DEVICE_Init()` is run, which makes them redundant.

Reply:
> used to be called in usbd_conf.c

I'm using a copy of that file from the original MidiBoot repo, didn't regenerate a new one. Can try that and see what's going to be the difference. I wasn't sure that it didn't have any customization before.

Reply:
After changing bootloader to use up to date version of cube's USBD files, those commented out MX_USB_OTG_*_PCD_Init calls still cause firmware to hang.

Also, USB is still functional with them commented out. I think it just gets added because we don't setup USB audio middleware in cube, normally we don't call them in other projects AFAICT.